explorer/ast/ast_rtti.h

```diff
@@ -37,82 +37,51 @@ class AstRttiNodeKind : public CARBON_ENUM_BASE(AstRttiNodeKind) {
   // other enumerations to match, and to implement range checks.
   using EnumBase::AsInt;
 
-  CARBON_AST_RTTI_KINDS(IGNORE, CARBON_ENUM_CONSTANT_DECLARATION)
+  CARBON_AST_RTTI_KINDS(CARBON_IGNORE, CARBON_ENUM_CONSTANT_DECLARATION)
 };
+
+// Define the constant members for AstRttiNodeKind.
 #define CONSTANT_DEFINITION(E) \
   CARBON_ENUM_CONSTANT_DEFINITION(AstRttiNodeKind, E)
-CARBON_AST_RTTI_KINDS(IGNORE, CONSTANT_DEFINITION)
+CARBON_AST_RTTI_KINDS(CARBON_IGNORE, CONSTANT_DEFINITION)
 #undef CONSTANT_DEFINITION
 
 // Define Kind enumerations for all base classes.
-#define DEFINE_ENUM(C)                                                      \
-  CARBON_DEFINE_RAW_ENUM_CLASS(C##Kind, int) {                              \
-    CARBON_##C##_KINDS(IGNORE, ENUMERATOR)                                  \
-  };                                                                        \
-  template <typename Derived>                                               \
-  class C##KindTemplate : public CARBON_ENUM_BASE_CRTP(C##Kind, Derived) {  \
-   private:                                                                 \
-    using Base = CARBON_ENUM_BASE_CRTP(C##Kind, Derived);                   \
-    friend class AstRttiNodeKind;                                           \
-                                                                            \
-   public:                                                                  \
-    using IsCarbonAstRttiNodeKind = void;                                   \
-                                                                            \
-    C##KindTemplate() = default;                                            \
-                                                                            \
-    /* This type can be explicitly converted from the generic node kind. */ \
-    explicit C##KindTemplate(AstRttiNodeKind base_kind)                     \
-        : Base(Base::FromInt(base_kind.AsInt())) {}                         \
-                                                                            \
-    CARBON_##C##_KINDS(IGNORE, CARBON_INLINE_ENUM_CONSTANT_DEFINITION)      \
-  };                                                                        \
-  class C##Kind : public C##KindTemplate<C##Kind> {                         \
-    using C##KindTemplate<C##Kind>::C##KindTemplate;                        \
+#define DEFINE_KIND_ENUM(C)                                                   \
```

Comment:
Should this be undefed? Maybe you renamed this one but not the #undef DEFINE_ENUM?

Reply:
Sorry, pushed an incomplete commit. PTAL :)